diff --git a/meson.build b/meson.build
index 72bc7b1..88bd657 100644
--- a/meson.build
+++ b/meson.build
@@ -23,10 +23,12 @@ endif
 
 cpu = host_machine.cpu_family()
 
-if cpu == 'x86_64'
-	c_args += '-m' + get_option('x86_64-simd')
-elif cpu == 'arm'
-	c_args += '-mfpu=neon'
+if get_option('cpu_optimize')
+	if cpu == 'x86_64'
+		c_args += '-m' + get_option('x86_64-simd')
+	elif cpu == 'arm'
+		c_args += '-mfpu=neon'
+	endif
 endif
 
 add_project_arguments(c_args, language: 'c')
@@ -39,13 +41,7 @@ pixman = dependency('pixman-1')
 libturbojpeg = dependency('libturbojpeg', required: get_option('tight-encoding'))
 gnutls = dependency('gnutls', required: get_option('tls'))
 zlib = dependency('zlib')
-
-aml_project = subproject('aml', required: false)
-if aml_project.found()
-	aml = aml_project.get_variable('aml_dep')
-else
-	aml = dependency('aml')
-endif
+aml = dependency('aml')
 
 inc = include_directories('include')
 
@@ -82,7 +78,7 @@ if gnutls.found()
 	config.set('ENABLE_TLS', true)
 endif
 
-if host_system == 'linux' and cc.has_header('sys/sdt.h')
+if host_system == 'linux' and get_option('systemtap')
 	config.set('HAVE_USDT', true)
 endif
 
diff --git a/meson_options.txt b/meson_options.txt
index 1c2eff6..359cf88 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -2,5 +2,9 @@ option('benchmarks', type: 'boolean', value: false, description: 'Build benchmar
 option('examples', type: 'boolean', value: false, description: 'Build examples')
 option('tight-encoding', type: 'feature', value: 'auto', description: 'Enable Tight encoding')
 option('tls', type: 'feature', value: 'auto', description: 'Enable encryption & authentication')
+option('cpu_optimize', type: 'boolean', value: false,
+       description: 'Enable cpu specific optimizations')
 option('x86_64-simd', type: 'string', value: 'sse2',
        description: 'Choose SIMD extension for x86_64 release build')
+option('systemtap', type: 'boolean', value: false, 
+       description: 'Enable tracing using sdt')
