diff --git a/SNI_sub.pro b/SNI_sub.pro
new file mode 100644
index 0000000..9c83351
--- /dev/null
+++ b/SNI_sub.pro
@@ -0,0 +1,4 @@
+
+TEMPLATE = subdirs
+
+SUBDIRS += Widgets/SNI
diff --git a/Widgets/SNI/SNI.pro b/Widgets/SNI/SNI.pro
index a3de1cb..102ae1e 100644
--- a/Widgets/SNI/SNI.pro
+++ b/Widgets/SNI/SNI.pro
@@ -20,8 +20,6 @@ HEADERS += statusnotifierbutton.h
 HEADERS += statusnotifieriteminterface.h
 HEADERS += statusnotifierwatcher.h
 HEADERS += statusnotifierwidget.h
-HEADERS += statusnotifieritem_adaptor.h
-HEADERS += statusnotifieritem_interface.h

 SOURCES += dbustypes.cpp
 SOURCES += main.cpp
@@ -30,8 +28,6 @@ SOURCES += statusnotifierbutton.cpp
 SOURCES += statusnotifieriteminterface.cpp
 SOURCES += statusnotifierwatcher.cpp
 SOURCES += statusnotifierwidget.cpp
-SOURCES += statusnotifieritem_adaptor.cpp
-SOURCES += statusnotifieritem_interface.cpp

 CONFIG += silent
