
index 92c6cdf..aa3cd18 100644
--- a/numba/tests/test_record_dtype.py
+++ b/numba/tests/test_record_dtype.py
@@ -778,7 +778,7 @@ class TestRecordDtype(unittest.TestCase):
         self.assertNotIn('first', transformed)
         self.assertNotIn('second', transformed)
         # Length is usually 50 - 5 chars tolerance as above.
-        self.assertLess(len(transformed), 50)
+        self.assertLessEqual(len(transformed), 50)
 
     def test_record_two_arrays(self):
         """
