diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9d8c1f9..e5d156b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,6 @@ SET(LIB_NAME "nuklear")
 ADD_LIBRARY(
 	"${LIB_NAME}"
 	MODULE
-	src/nuklear/nuklear.h
 	src/nuklear_love.c
 )
 
@@ -24,6 +23,8 @@ TARGET_LINK_LIBRARIES(
 	${LUA_LIBRARIES}
 )
 
+include(GNUInstallDirs)
+
 IF(MSVC)
   TARGET_COMPILE_DEFINITIONS(${LIB_NAME} PRIVATE LUA_BUILD_AS_DLL)
 endif(MSVC)
@@ -34,4 +35,4 @@ IF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
   SET(CMAKE_INSTALL_PREFIX "${CMAKE_BINARY_DIR}" CACHE PATH "..." FORCE)
 ENDIF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 
-INSTALL(TARGETS "${LIB_NAME}" DESTINATION .)
+INSTALL(TARGETS "${LIB_NAME}" DESTINATION ${CMAKE_INSTALL_LIBDIR})
diff --git a/src/nuklear_love.c b/src/nuklear_love.c
index 0ca9213..728abd3 100644
--- a/src/nuklear_love.c
+++ b/src/nuklear_love.c
@@ -26,7 +26,7 @@
 #define NK_VECTOR_STACK_SIZE 128
 #define NK_FLAGS_STACK_SIZE 64
 #define NK_COLOR_STACK_SIZE 256
-#include "nuklear/nuklear.h"
+#include "nuklear.h"
 
 /*
  * ===============================================================
